custom_components/bmw_connected_drive/sensor.py

```diff
@@ -66,21 +161,55 @@ async def async_setup_entry(hass, config_entry, async_add_entities):
                     account, vehicle, attribute_name, attribute_info
                 )
                 devices.append(device)
+        # LastTrip
+        if vehicle.has_statistics_service:
+            for attribute_name in ATTR_LAST_TRIP:
+                device = BMWConnectedDriveSensor(
+                    account, vehicle, attribute_name, attribute_info, LAST_TRIP
+                    )
+                devices.append(device)
+        # AllTrips
+        if vehicle.has_statistics_service:
```

Comment:
This is the same check as on line 165, so not needed to do this twice.

Can't you use this function to determine what services to use here-> https://github.com/bimmerconnected/bimmer_connected/blob/760511803d069f0637de9ded817ae2a95f424d83/bimmer_connected/vehicle.py#L190

That also gives SERVICE_RANGEMAP, SERVICE_EFFICIENCY and SERVICE_NAVIGATION which can be added for certain cars.